### PR TITLE
[CDAP-4060] modify workflow status to poll from single API

### PR DESCRIPTION
**Pull request by elfenheart**
~~TODO: need to test with an App with Custom Action~~

### CODE REVIEW

**Comment by bdmogal** [comment]
FYI: You can use the app from https://github.com/cdap-guides/cdap-workflow-guide as an example of a workflow with a custom action.

**Comment by elfenheart** [comment]
Tested with Custom Action. 

screen shot 2016-04-12 at 11 24 51 am

File: cdap-ui/app/features/workflows/controllers/tabs/runs/tabs/status-ctrl.js [hunk 1]
```diff
@@ -100,53 +100,21 @@ class WorkflowsRunsStatusController {
...
-
-            this.mySparkApi.runDetail(sparkParams)
-              .$promise
-              .then( (result) => {
-                this.data.current[node.name] = result.status;
-                this.onChangeFlag += 1;
-              });
-          }
-        });
+        this.myWorkFlowApi.getNodesState(runparams)
```
Comment (chtyim) [inline cdap-ui/app/features/workflows/controllers/tabs/runs/tabs/status-ctrl.js]: What a simplification. :+1:

**Comment by chtyim** [comment]
LGTM

**Pull request merged**